Lib/configparser.py

```diff
@@ -687,7 +688,7 @@ def read(self, filenames, encoding=None):
 
         Return list of successfully read files.
         """
-        if isinstance(filenames, str):
+        if isinstance(filenames, (str, bytes, os.PathLike)):
```

Comment:
I've made a new bug report here:
http://bugs.python.org/issue29627

I think it may be that this fix should come after that fix as I'm guessing it may affect python versions without pathlib? (If support for bytes is added).